### PR TITLE
Adds Persistent Volume to Aqueduct in-cluster Deployment

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Adds Persistent Volume to Aqueduct in-cluster Deployment.
## Related issue number (if any)
https://linear.app/aqueducthq/issue/ENG-1801/add-persistent-volume-to-helm-chart
## Loom demo (if any)
```
(base) ubuntu@ip-172-31-46-200:~$ kubectl get pv -n aqueduct
NAME          CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                        STORAGECLASS   REASON   AGE
aqueduct-pv   10Gi       RWO            Retain           Bound    aqueduct/aqueduct-pv-claim   manual                  5m50s
```
in description of the pod:
```
Volumes:
  aqueduct-pv-storage:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  aqueduct-pv-claim
    ReadOnly:   false
```

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/aqueduct-helm/templates/persistentvolume.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: aqueduct-pv
+spec:
+  storageClassName: aqueduct
+  capacity:
+    storage: 5Gi
+  accessModes:
+    - ReadWriteOnce
+  hostPath:
+    path: "/root/.aqueduct"
```
Comment (cw75) [inline src/aqueduct-helm/templates/persistentvolume.yaml:12]: It seems from here that `hostPath` is only used for dev and testing for single-node cluster? Does this work for multi-node clusters?

**Review by cw75 — COMMENTED** [review]
@hsubbaraj-spiral Left a couple questions inline. Have we tested that if we kill the pod and restart another one, data in `~/.aqueduct` is persisted?

File: src/aqueduct-helm/templates/persistentvolume.yaml (hunk 1 above)
Comment (hsubbaraj-spiral) [reply src/aqueduct-helm/templates/persistentvolume.yaml:12]: Yep I tested this on multi-node clusters as well.

File: src/aqueduct-helm/templates/persistentvolumeclaim.yaml [hunk 2]
```diff
@@ -3,8 +3,9 @@ kind: PersistentVolumeClaim
 metadata:
```
[the remark this answers is not in the document]
Comment (hsubbaraj-spiral) [reply src/aqueduct-helm/templates/persistentvolumeclaim.yaml:3]: The persistent volume claim is sufficed by the persistent volume that we generate. There isn't a direct link, that resource just needs to be available to claim.

**Review by cw75 — APPROVED** [review]
LGTM! Can you also add a link to the documentation of how dynamic provisioning works?

Another thing is that I wonder if we should mount just `/root/.aqueduct` or the whole `/root/*`. Because the server pod may have installed some Python dependencies that's not gonna be stored in `/root/.aqueduct`, so if we only mount this dir we will lose these dependency installations.

**Pull request merged**